src/dishka/integrations/typer.py

```diff
@@ -7,27 +7,109 @@
 ]
 
 from collections.abc import Callable
-from typing import Final, TypeVar
+from inspect import Parameter
+from typing import Final, ParamSpec, TypeVar, cast, get_type_hints
 
+import click
 import typer
-from click import get_current_context
 
-from dishka import Container, FromDishka
+from dishka import Container, FromDishka, Scope
+from dishka.dependency_source.make_context_var import from_context
+from dishka.provider import Provider
 from .base import is_dishka_injected, wrap_injection
 
 T = TypeVar("T")
+P = ParamSpec("P")
 CONTAINER_NAME: Final = "dishka_container"
 
 
-def inject(func: Callable[..., T]) -> Callable[..., T]:
-    return wrap_injection(
-        func=func,
-        container_getter=lambda _, __: get_current_context().meta[
-            CONTAINER_NAME
-        ],
-        remove_depends=True,
-        is_async=False,
+def inject(func: Callable[P, T]) -> Callable[P, T]:
```

Comment:
Please, do not duplicate code of `wrap_injection`, just call it with proper params

Reply:
Okay, I found a way! :)